# ADS 505 Final Project

## Best Home Regions

### Claire Bentzen, Jiaqi He, S M Sultan Mahmud Rahat

## 1. Data Preprocessing

In [1]:
import pandas as pd

In [2]:
# read in data
homes = pd.read_csv('final_dataset.csv')
homes.head()

MortgageRateConventionalFixed_2011-06  \
0                               4.374217   
1                               4.374217   
2                               4.374217   
3                               4.374217   
4                               4.374217   

   MortgageRateConventionalFixed_2011-07  \
0                               4.460115   
1                               4.460115   
2                               4.460115   
3                               4.460115   
4                               4.460115   

   MortgageRateConventionalFixed_2011-08  \
0                               4.127827   
1                               4.127827   
2                               4.127827   
3                               4.127827   
4                               4.127827   

   MortgageRateConventionalFixed_2011-09  \
0                               3.963005   
1                               3.963005   
2                               3.963005   
3                               3.963005   
4                               3.963005   

   MortgageRateConventionalFixed_2011-10  \
0                               4.035497   
1                               4.035497   
2                               4.035497   
3                               4.035497   
4                               4.035497   

   MortgageRateConventionalFixed_2011-11  \
0                               3.941695   
1                               3.941695   
2                               3.941695   
3                               3.941695   
4                               3.941695   

   MortgageRateConventionalFixed_2011-12  \
0                                3.85004   
1                                3.85004   
2                                3.85004   
3                                3.85004   
4                                3.85004   

   MortgageRateConventionalFixed_2012-01  \
0                               3.815944   
1                               3.815944   
2                               3.815944   
3                               3.815944   
4                               3.815944   

   MortgageRateConventionalFixed_2012-02  \
0                               3.808312   
1                               3.808312   
2                               3.808312   
3                               3.808312   
4                               3.808312   

   MortgageRateConventionalFixed_2012-03  ...  PriceToRentRatio_2017-08  \
0                               3.931269  ...                     10.64   
1                               3.931269  ...                     12.94   
2                               3.931269  ...                     11.36   
3                               3.931269  ...                     12.94   
4                               3.931269  ...                     11.98   

   PriceToRentRatio_2017-09  RegionID  RegionName                 City  State  \
0                     10.72     58196        1001               Agawam     MA   
1                     13.15     58197        1002              Amherst     MA   
2                     11.50     58200        1005        Town of Barre     MA   
3                     12.77     58201        1007  Town of Belchertown     MA   
4                     11.97     58202        1008            Blandford     MA   

         Metro  CountyName  SizeRank  New_Composite_Score  
0  Springfield     Hampden      5770             0.243539  
1  Springfield   Hampshire      4112             0.317422  
2    Worcester   Worcester     11428             0.267254  
3  Springfield   Hampshire      6855             0.305868  
4  Springfield     Hampden     15088             0.281545  

[5 rows x 168 columns]

In [3]:
# identify number of rows and columns
homes.shape

(14233, 168)

#### Check for Duplicate Records

In [4]:
# check for duplicates
duplicate_homes = homes[homes.duplicated()]
duplicate_homes.shape

(0, 168)

There are no duplicate records in the dataset.

#### Check for Missing Values

In [5]:
# check for overall percentage of missing values
missing_values = homes.isnull().sum().sum()
percent_missing = (missing_values / homes.size) * 100
print("Total percentage of missing data: {:.2f}%".format(percent_missing))

Total percentage of missing data: 1.63%


In [6]:
pd.set_option('display.max_rows', None)

# check which columns missing values reside in
missing_homes = homes.isnull().sum()
columns_missing_values = missing_homes[missing_homes > 0]

# turn missing values series into dataframe
columns_missing_values_df = columns_missing_values.to_frame()
columns_missing_values_df.columns = ['Count']

In [7]:
# missing values as percentages
columns_missing_values_percents = (columns_missing_values / homes.size) * 100

# turn missing values percentage series into dataframe
columns_missing_values_percents_df = columns_missing_values_percents.to_frame()
columns_missing_values_percents_df.columns = ['%']

In [8]:
# merge dataframes to very counts against percentages
missing = columns_missing_values_df.join(columns_missing_values_percents_df, how='inner')
missing

Count         %
PriceToRentRatio_2010-10    994  0.041570
PriceToRentRatio_2010-11   1184  0.049516
PriceToRentRatio_2010-12    906  0.037890
PriceToRentRatio_2011-01    691  0.028898
PriceToRentRatio_2011-02    651  0.027225
PriceToRentRatio_2011-03    724  0.030278
PriceToRentRatio_2011-04    635  0.026556
PriceToRentRatio_2011-05    644  0.026933
PriceToRentRatio_2011-06    693  0.028982
PriceToRentRatio_2011-07    632  0.026431
PriceToRentRatio_2011-08    608  0.025427
PriceToRentRatio_2011-09    609  0.025469
PriceToRentRatio_2011-10    607  0.025385
PriceToRentRatio_2011-11    645  0.026975
PriceToRentRatio_2011-12    683  0.028564
PriceToRentRatio_2012-01    633  0.026473
PriceToRentRatio_2012-02    591  0.024716
PriceToRentRatio_2012-03    617  0.025804
PriceToRentRatio_2012-04    617  0.025804
PriceToRentRatio_2012-05    632  0.026431
PriceToRentRatio_2012-06    679  0.028396
PriceToRentRatio_2012-07    664  0.027769
PriceToRentRatio_2012-08    701  0.029317
PriceToRentRatio_2012-09    636  0.026598
PriceToRentRatio_2012-10    670  0.028020
PriceToRentRatio_2012-11    602  0.025176
PriceToRentRatio_2012-12    619  0.025887
PriceToRentRatio_2013-01    628  0.026264
PriceToRentRatio_2013-02    608  0.025427
PriceToRentRatio_2013-03    641  0.026807
PriceToRentRatio_2013-04    579  0.024214
PriceToRentRatio_2013-05    529  0.022123
PriceToRentRatio_2013-06    504  0.021078
PriceToRentRatio_2013-07    505  0.021120
PriceToRentRatio_2013-08    446  0.018652
PriceToRentRatio_2013-09    446  0.018652
PriceToRentRatio_2013-10    436  0.018234
PriceToRentRatio_2013-11    435  0.018192
PriceToRentRatio_2013-12    472  0.019740
PriceToRentRatio_2014-01    471  0.019698
PriceToRentRatio_2014-02    457  0.019112
PriceToRentRatio_2014-03    454  0.018987
PriceToRentRatio_2014-04    437  0.018276
PriceToRentRatio_2014-05    467  0.019530
PriceToRentRatio_2014-06    404  0.016896
PriceToRentRatio_2014-07    406  0.016979
PriceToRentRatio_2014-08    404  0.016896
PriceToRentRatio_2014-09    404  0.016896
PriceToRentRatio_2014-10    404  0.016896
PriceToRentRatio_2014-11    404  0.016896
PriceToRentRatio_2014-12    406  0.016979
PriceToRentRatio_2015-01    414  0.017314
PriceToRentRatio_2015-02    401  0.016770
PriceToRentRatio_2015-03    401  0.016770
PriceToRentRatio_2015-04    413  0.017272
PriceToRentRatio_2015-05    401  0.016770
PriceToRentRatio_2015-06    402  0.016812
PriceToRentRatio_2015-07    435  0.018192
PriceToRentRatio_2015-08    397  0.016603
PriceToRentRatio_2015-09    397  0.016603
PriceToRentRatio_2015-10    396  0.016561
PriceToRentRatio_2015-11    391  0.016352
PriceToRentRatio_2015-12    383  0.016017
PriceToRentRatio_2016-01    383  0.016017
PriceToRentRatio_2016-02    413  0.017272
PriceToRentRatio_2016-03    396  0.016561
PriceToRentRatio_2016-04    389  0.016268
PriceToRentRatio_2016-05    387  0.016185
PriceToRentRatio_2016-06    272  0.011375
PriceToRentRatio_2016-07    218  0.009117
PriceToRentRatio_2016-08    255  0.010664
PriceToRentRatio_2016-09    217  0.009075
PriceToRentRatio_2016-10    217  0.009075
PriceToRentRatio_2016-11    217  0.009075
PriceToRentRatio_2016-12    260  0.010873
PriceToRentRatio_2017-01    216  0.009033
PriceToRentRatio_2017-02    214  0.008950
PriceToRentRatio_2017-03    213  0.008908
PriceToRentRatio_2017-04     38  0.001589

In [9]:
# drop missing values in the dataset
homes = homes.dropna()
homes.shape

(12786, 168)

Although missing values only made up 1.63% of the overall dataset, we have dropped rows that contained missing values which brought the dataset to have 12,786 remaining records.

## 2. Exploratory Data Analysis